### PR TITLE
feat: extract android applicationId

### LINKED ISSUE #1884: [Android] Derive default app id from application id, not package name
## Describe the Feature
In most cases the application id and package name / namespace are the same string. However in some cases they are not. For example, if a project got renamed but didn't want to change all their native code, or they ship with different app ids for prod, testing, local development, etc. This raises the question: why is the default app id (if you don't explicitly pass `--appId`) based on the package name and not the actual application id?

For example, if I have a React Native 0.71 project with the `android/app/build.gradle` configured with

```
android {
    namespace "com.example.foo"
    defaultConfig {
        applicationId: "com.example.bar"
    }
}
```

then running `react-native run-android` with no other flags will successfully build and install on the device, but will fail to start because it's trying to run the app `com.example.foo`.

I think a much more sensible default value would be to use the actual resolved `applicationId` value.

### PR DESCRIPTION
**Pull request by adamTrz**
Summary:
---------

Use `applicationId` specified in `defaultConfig` of `build.grade` file. 

Fixes #1884 

Test Plan:
----------

1. Clone the fork and checkout to current branch
2. Follow [Contributing guide](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md) (link CLI)
3. Change default `applicationId` inside `defaultConfig` of `build.gradle` file, eg.
```diff
 namespace "com.foo"
    defaultConfig {
+        applicationId "com.bar"
-        applicationId "com.foo"
```
4. Run the app

----------

TODO's / things to consider 🤔 :
- [ ] we do not always have `appld` present at `defaultConfig` object in `build.gradle` file. Should we default then to namespace?
- [ ] users can have different applicationId's specified for build variants / flavors. Do we want to cover those cases as well?

### CODE REVIEW

File: packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts [hunk 1]
```diff
@@ -13,11 +13,12 @@ import {logger, CLIError} from '@react-native-community/cli-tools';
 function tryLaunchAppOnDevice(
   device: string | void,
   packageName: string,
+  applicationId: string,
   adbPath: string,
   args: Flags,
 ) {
   const {appId, appIdSuffix} = args;
-  const packageNameWithSuffix = [appId || packageName, appIdSuffix]
+  const packageNameWithSuffix = [appId || applicationId, appIdSuffix]
```
Comment (cortinico) [inline packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts:21]: variable name is confusign here as it's effectively the applicationID then

File: packages/cli-platform-android/src/config/__fixtures__/files/build.gradle [hunk 2]
```diff
@@ -3,6 +3,9 @@ apply package: "com.facebook.react"
 
 android {
     namespace 'com.some.example'
```
Comment (cortinico) [inline packages/cli-platform-android/src/config/__fixtures__/files/build.gradle:5]: nit: let's use double quotes here as well

**Review by cortinico — APPROVED** [review]
LGTM just a couple of nits

File: packages/cli/src/index.ts [hunk 3]
```diff
@@ -16,8 +16,8 @@ const pkgJson = require('../package.json');
 const program = new CommanderCommand()
   .usage('[command] [options]')
   .version(pkgJson.version, '-v', 'Output the current version')
-  .option('--verbose', 'Increase logging verbosity')
-  .enablePositionalOptions();
+  .option('--verbose', 'Increase logging verbosity');
+// .enablePositionalOptions();
```
Comment (cortinico) [inline packages/cli/src/index.ts]: did you intend to remove this line?

Comment (adamTrz) [reply packages/cli/src/index.ts]: Nah. That should be reverted 👌

**Review by TMisiukiewicz — APPROVED** [review]

**Comment by Tchaikovsky1114** [comment]
thank you for great works!

how to use that ?
i have problem with this issue like different namespace to applicationId..

**Pull request merged**